salt/utils/cache.py

```diff
@@ -345,3 +348,32 @@ def context_cache_wrap(*args, **kwargs):
         return func(*args, **kwargs)
 
     return context_cache_wrap
+
+
+def verify_cache_version(cache_path):
+    """
+    Check that cache version matches salt version.
+    If cache version deos not matches salt version wipe the cache.
```

Comment:
Check that the cached version matches the Salt version.
If the cached version does not match the Salt version, wipe the cache.